repository.go

```diff
@@ -196,15 +199,25 @@ func PlainOpen(path string) (*Repository, error) {
 	var wt, dot billy.Filesystem
 
 	fs := osfs.New(path)
-	if _, err := fs.Stat(".git"); err != nil {
+	fi, err := fs.Stat(".git")
+	if err != nil {
```

Comment:
Separating it into such functions would require doing redundant stats operations, or passing quite different arguments to each function. Could you review the new version and check if it's more readable now?